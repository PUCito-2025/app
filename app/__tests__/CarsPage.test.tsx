import { render } from "@testing-library/react";
import { expect, test } from "vitest";

import CarDetailsPage from "@/app/cars/[id]/page";

test("CarsPage matches snapshot", () => {
  const { container } = render(<CarDetailsPage />);
  expect(container).toMatchSnapshot();
});
