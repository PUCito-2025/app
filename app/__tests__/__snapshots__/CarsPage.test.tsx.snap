// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CarsPage matches snapshot 1`] = `
<div>
  <div
    class="space-y-10 px-4 py-6 lg:px-10"
  >
    <div
      class="relative w-full overflow-hidden rounded-md shadow"
    >
      <div
        class="h-64 w-full bg-gray-200"
      />
      <div
        class="absolute bottom-4 left-6 text-white"
      >
        <h1
          class="text-2xl font-bold"
        >
          Mercedes-Benz C200 2023
        </h1>
        <p
          class="text-sm"
        >
          Santiago, Chile • 18,000 km
        </p>
      </div>
      <div
        class="absolute right-6 bottom-4 text-xl font-semibold text-white"
      >
        CLP $35,990,000
      </div>
    </div>
    <div
      class="grid grid-cols-1 gap-6 lg:grid-cols-2"
    >
      <div
        class="space-y-3 rounded-md border bg-white p-6"
      >
        <h2
          class="mb-4 text-lg font-semibold"
        >
          Vehicle Specifications
        </h2>
        <div
          class="grid grid-cols-2 gap-3 text-sm"
        >
          <p>
            <strong>
              Brand/Model:
            </strong>
             Mercedes-Benz C200
          </p>
          <p>
            <strong>
              Color:
            </strong>
             Silver
          </p>
          <p>
            <strong>
              Year:
            </strong>
             2023
          </p>
          <p>
            <strong>
              Engine:
            </strong>
             2.0L Turbo
          </p>
          <p>
            <strong>
              Transmission:
            </strong>
             Automatic
          </p>
          <p>
            <strong>
              VIN:
            </strong>
             WDD2040**********
          </p>
          <p>
            <strong>
              Fuel Type:
            </strong>
             Petrol
          </p>
        </div>
        <button
          class="mt-4 rounded-md bg-blue-600 px-4 py-2 text-sm text-white hover:bg-blue-700"
          type="button"
        >
          Check Vehicle History
        </button>
      </div>
      <div
        class="rounded-md border bg-white p-6"
      >
        <h2
          class="mb-4 text-lg font-semibold"
        >
          Seller Notes
        </h2>
        <p
          class="text-sm text-gray-700"
        >
          Excelente estado general, mantenido en taller oficial. Inspección técnica al día, sin siniestros reportados. Interior como nuevo, se entrega con mantención recién hecha. Llantas originales y neumáticos nuevos.
        </p>
      </div>
    </div>
    <div
      class="rounded-md border bg-white p-6"
    >
      <h2
        class="mb-4 text-lg font-semibold"
      >
        Available On
      </h2>
      <div
        class="grid grid-cols-1 gap-4 text-sm md:grid-cols-3"
      >
        <div
          class="space-y-2 rounded border p-4"
        >
          <p
            class="font-medium"
          >
            Yapo
          </p>
          <p
            class="font-bold text-blue-600"
          >
            $35,990,000
          </p>
          <p
            class="text-gray-600"
          >
            Seller: 
            Juan Pérez
          </p>
          <p
            class="text-gray-600"
          >
            Location: 
            Santiago
          </p>
          <p
            class="text-xs text-gray-500"
          >
            3 hours ago
          </p>
          <button
            class="mt-2 w-full rounded-md border py-1 text-sm hover:bg-gray-100"
            type="button"
          >
            View Original
          </button>
        </div>
        <div
          class="space-y-2 rounded border p-4"
        >
          <p
            class="font-medium"
          >
            Facebook
          </p>
          <p
            class="font-bold text-blue-600"
          >
            $36,700,000
          </p>
          <p
            class="text-gray-600"
          >
            Seller: 
            María González
          </p>
          <p
            class="text-gray-600"
          >
            Location: 
            Temuco
          </p>
          <p
            class="text-xs text-gray-500"
          >
            6 hours ago
          </p>
          <button
            class="mt-2 w-full rounded-md border py-1 text-sm hover:bg-gray-100"
            type="button"
          >
            View Original
          </button>
        </div>
        <div
          class="space-y-2 rounded border p-4"
        >
          <p
            class="font-medium"
          >
            Chileautos
          </p>
          <p
            class="font-bold text-blue-600"
          >
            $39,000,000
          </p>
          <p
            class="text-gray-600"
          >
            Seller: 
            Auto Import SPA
          </p>
          <p
            class="text-gray-600"
          >
            Location: 
            Santiago
          </p>
          <p
            class="text-xs text-gray-500"
          >
            5 hours ago
          </p>
          <button
            class="mt-2 w-full rounded-md border py-1 text-sm hover:bg-gray-100"
            type="button"
          >
            View Original
          </button>
        </div>
      </div>
    </div>
    <div
      class="grid grid-cols-1 gap-6 lg:grid-cols-2"
    >
      <div
        class="flex h-64 items-center justify-center rounded-md border bg-white p-6"
      >
        <p
          class="text-sm text-gray-500"
        >
          [Price Chart Placeholder — last 60–90 days]
        </p>
      </div>
      <div
        class="rounded-md border bg-white p-6"
      >
        <h2
          class="mb-4 text-lg font-semibold"
        >
          Similar Vehicles
        </h2>
        <ul
          class="space-y-3 text-sm"
        >
          <li
            class="flex items-start justify-between"
          >
            <div>
              <p
                class="font-medium"
              >
                Mercedes-Benz C200 2022
              </p>
              <p
                class="text-xs text-gray-500"
              >
                12,000 km
              </p>
            </div>
            <p
              class="font-bold text-blue-600"
            >
              $29,900,000
            </p>
          </li>
          <li
            class="flex items-start justify-between"
          >
            <div>
              <p
                class="font-medium"
              >
                Mercedes-Benz C200 2019
              </p>
              <p
                class="text-xs text-gray-500"
              >
                18,000 km
              </p>
            </div>
            <p
              class="font-bold text-blue-600"
            >
              $24,900,000
            </p>
          </li>
          <li
            class="flex items-start justify-between"
          >
            <div>
              <p
                class="font-medium"
              >
                Mercedes-Benz C200 2020
              </p>
              <p
                class="text-xs text-gray-500"
              >
                15,000 km
              </p>
            </div>
            <p
              class="font-bold text-blue-600"
            >
              $35,500,000
            </p>
          </li>
        </ul>
        <button
          class="mt-4 w-full rounded-md border py-1 text-sm hover:bg-gray-100"
          type="button"
        >
          Compare Selected
        </button>
      </div>
    </div>
  </div>
</div>
`;
